src/Provider/ImageProvider.php

```diff
@@ -215,9 +216,28 @@ protected function doTransform(MediaInterface $media)
             return;
         }
 
-        if (!\in_array(strtolower(pathinfo($fileName, \PATHINFO_EXTENSION)), $this->allowedExtensions, true)
-            || !\in_array($media->getBinaryContent()->getMimeType(), $this->allowedMimeTypes, true)) {
-            return;
+        $extension = strtolower(pathinfo($fileName, \PATHINFO_EXTENSION));
+
+        if (!\in_array($extension, $this->allowedExtensions, true)) {
+            $media->setProviderStatus(MediaInterface::STATUS_ERROR);
+
+            throw new UploadException(sprintf(
+                'The image extension "%s" is not one of the allowed (%s)',
+                $extension,
+                implode(', ', $this->allowedExtensions)
```

Comment:
```suggestion
                '"'.implode('", "', $this->allowedExtensions).'"'
```

Reply:
This would produce "" for an empty array, if we want to add quotes it needs more thinking

Reply:
If this case is possible, I think we could add a previous condition (`[] === $this->allowedExtensions`) with a specific message stating that there are no allowed extensions.

Reply:
Fixed.